lib/resty/healthcheck.lua

```diff
@@ -1236,18 +1236,14 @@ local function checker_callback(self, health_mode)
   if not list_to_check[1] then
     self:log(DEBUG, "checking ", health_mode, " targets: nothing to do")
   else
-    local timer = resty_timer({
-      interval = 0,
-      recurring = false,
-      immediate = false,
-      detached = true,
-      expire = function()
-        self:log(DEBUG, "checking ", health_mode, " targets: #", #list_to_check)
-        self:active_check_targets(list_to_check)
-      end,
-    })
-    if timer == nil then
-      self:log(ERR, "failed to create timer to check ", health_mode)
+    local thread = ngx.thread.spawn(function(mode, list)
```

Comment:
I do not agree with this change
`checker_callback()` was called by `active_check_timer` timer function, that function for loop will iterate all active health check instances, and then submit an active health checker task by timer, that's our existing implementation. But After this change, the health checker task from `timer` to `ngx.thread` , which means from async thread to sync thread. In async mode, we submit the health checker task and then do not wait for the task return status, but in sync thread, we need to wait task return thread, So every active healthcheck instance for the loop needs to wait for all health check tasks done, That's not accepted for us when we have many upstream node need to active healthchecker eg: if some upstream nodes can not connect, it will trigger timeout logic, the whole for loop execution time was sum by all node connection timeout.

Reply:
A possible solution to this problem is that when we create an active health check task, we check the completion status of the previous task on this instance. If it is still in progress, we cancel the submission of the current task, but this may be a break change in bad case.

Reply:
@oowl I created an alternative solution for this: https://github.com/Kong/lua-resty-healthcheck/pull/156

To keep the behaviour as much similar as before, I checked the `running status` of previous task on this instance, instead of the `completion status`. The behavior of checking the `completion status` will be just the same as the current PR which waits for the coroutine to finish.